### ISSUE TITLE
l10n_ec

### ISSUE THREAD

**Issue by lap-odoo** [odoo/odoo#20563]
Dear Odoo Partners,

At Odoo, a lot of effort is put into improving the accounting localizations for each country and also to maintain these accounting localizations up-to-date. To achieve that, we will need your help.

One github issue will be created per accounting localization (e.g. l10n_be for Belgium). We expect from you to make suggestions about what should be improved in the specific accounting localization. Do not hesitate to share files with detailed explanations.

As soon as changes will be implemented, you will be informed and will have the opportunity to test it and share your feedback.

This github issue will also be useful to keep you informed about all the ongoing improvements concerning a specific accounting localization.

Please note that it may take some time before we start working on your localization. It depends on our resources and on the priority of the task. But the sooner we collect the information, the better.

Thank you for your help

**Comment by damendieta** [comment]
Hi, thanks for opening this issue.

The first step will be to adapt the CoA to the legal chart in Ecuador.

We have a legal chart of accounts that is base for reports of SME's, we can find it here:

[http://appscvs.supercias.gob.ec/guiasUsuarios/images/guias/info_fin/otros/PLAN%20DE%20CUENTAS.pdf](Ecuador's chart of accounts) 

The accounts signaled with a D are the actual accounts while the ones marked with T are "view" accounts, now goups in Odoo 11.

Daniel.

**Comment by damendieta** [comment]
After that, we need to define our taxes, here we have 2 main forms to declare, 103 and 104, both are declared monthly or every six months, depending on the fiscal position of the company.

My suggestion is to name the tax by the field on the form, for example:

In the form 104 we have several taxes that have 0% of IVA so, define a "IVA 0%" or a "IVA 0% Compras" (for purchases) is not useful, as we have not enough information to make the corresponding declarations.

So, a tax called 401 will be referenced to the field 401 in the form 104 and a 344 will be for the field 344 in the form 103, really few fields can get confused, like the field 500 in the form 104 will conflict with the field 500 in the form 103, but we can fix that naming them like 500 IVA and 500 IR, or, 500 104 and 500 103, or something like that.

**Comment by lap-odoo** [comment]
Thank you for the information. 

In Odoo, there is already a CoA for l10n_ec, is this one outdated ? 

For the CoA, do you think you could provide us : 
- a csv with the account groups (= accounts of type view), I need columns for : Name / Code Prefix / Parent
- a csv with the accounts (= the CoA), I need columns for : Code / Name / Type (using the standard accounts types that exist in Odoo) / Group (=account of type view) 

(If you are familiar with the PR, you can make one). 

For the taxes, we use the tags to define where the tax should be reported in the tax reports (103 and 104). Are the current taxes in Odoo correct ? They don't have the same names, so no confusion possible. However, the tags could be rename based on tax report you have to submit. 

Could you please provide more info about these tax reports ? 

Thanks

**Comment by damendieta** [comment]
Hi, yes, the CoA in the l10n_ec module is 5 or 6 years old, it's not accurate from v8.

I have the l10n_ec module already updated for v10 in the OCA repos, I'll update the module to work with v11 and make a PR

Daniel.

**Comment by lap-odoo** [comment]
Perfect, can you notify me when it's ready ? 

Thanks

**Comment by archimagogabo** [comment]
Hello, I'm a Silver Partner of Odoo Ecuador, we urgently need to have a complete and current  accounting localizations for Ecuador.
I am willing to put my team of accountants and everything that is necessary to give you all the requirements of accounting, billing, withholding, and all the requirements to integrate with Ecuador's electronic invoicing . let's work together to open up the fastest and most complete accounting localizations in Ecuador.
How would you like to set up a meeting on skype so that we can generate a work plan, know the formats and deliver the complete requirements?

**Comment by damendieta** [comment]
Hi, I have accounting, tax and e-invoicing already working for Ecuador.

I have it in V9, but we tested it on V11 past week and it worked out of the box, we haven't tested all, bit any change must be relatively easy.

We want to make it good enough to propose it to become part of Odoo, and we need help (we need to make it compliant with standars, make test, etc.).

Please, give me a call and let's talk about it.

My numbers are 0996601902 and 0996577156.

**Comment by damendieta** [comment]
Hi, yes, I can contribute all my code with no problem.

I have it for v9, so, it's already in a version out of support.

Most of my developments, if not all of them, are to be compliant with legal requirements, but the amount of code is too big to be post in one "big bang" pull request.

I can propose the following way to contribute.

1.- I can make a task in this issue or another issue with one required improvement, for example, create Ecuador's COA.
2.- I can upload the code I have that solves that problema.
3.- We check the code and make the improvements needed to be compliant with Odoo development standards and best practices.
4.- Merge the code.
5.- Go to the next issue (task).

Only one thing, all my contributions should be on the community version. Do you think it's possible?

@jcr-odoo What do you think?

**Comment by jcr-odoo** [comment]
Lets start with the PR for COA and Taxes, this shouldnt be a too big problem integrating those.
Could you list the other features that you have integratet like the e-invoice and what they do exaclty functionnally speaking? So that I can get a first insight 

For your request it will be as follows. If you want, you share your code with us, we review the functional aspect and the code, if we consider it good, we take over the code or write a code that is compliant with odoo. 
Its most likely to be community if it concerns apps that are already in the community version and if its a necessary improvement to make those apps work out of the box in community.

**Comment by damendieta** [comment]
They are mostly related with accounting, so, it's there for v12, but I don't know if it will be there for v13 too.

I'll make a PR for the COA someday on the next week, I would like to review it and add the groups staff, that is not there in v9.

It's ok if I make the PR for v12? or should it be for master?.

**Comment by jcr-odoo** [comment]
Master

**Comment by jcr-odoo** [comment]
@damendieta 

Hello,
I kindly would like to ask you if you have the time to make the PR? 

We could start investigating your module then

KR

**Comment by jco-odoo** [comment]
@damendieta If you want, we can use the information you gave us to already create a branch ourselves for the update of the CoA.  (it is a start)

**Comment by jco-odoo** [comment]
@damendieta For the moment, we are waiting for you.  (nothing urgent, but we appreciate your input)

**Comment by damendieta** [comment]
Hi, @jco-odoo sorry, I'm preparing a development environment and a group of people that want to cooperate with the localization of Ecuador.

Next week I'll be starting the migration of my modules (they are in V9) and all information will be available as public development, of course, I'll be making PR to this repository.

Only one thing, I'm making all my development efforts for v12, not master. 

How do you think we can manage that. Can you help me with the migration of the code to master?

Daniel.

**Comment by jco-odoo** [comment]
No problem for v12 as master can have breaking changes.  For what we get merged, we can handle the transition to master.

**Comment by celm1990** [comment]
@damendieta how I help with localization, you have roadmap or any document for shared?

**Comment by damendieta** [comment]
Hi @celm1990 the first thing is to make the l10n_ec account chart according to the new chart from the SUPERCIAS, this is the latest version (I know about).

[PLAN_CUENTAS_NUEVO_SISTEMA.xlsx](https://github.com/odoo/odoo/files/3092029/PLAN_CUENTAS_NUEVO_SISTEMA.xlsx)

We are coordinating our activities via a Telegram group, you can join us here.

https://t.me/odooec

The group is in Spanish, as 90% of the participants don't speak English.

**Comment by C3POdoo** [comment]
Dear @lap-odoo,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
